src/core/cfg.cpp

```diff
@@ -2502,6 +2502,32 @@ CFG* computeCFG(SourceInfo* source, std::vector<AST_stmt*> body) {
         }
     }
 
+    if (source->ast->type == AST_TYPE::FunctionDef || source->ast->type == AST_TYPE::Lambda) {
+        // Unpack tuple arguments
+        // Tuple arguments get assigned names ".0", ".1" etc. So this
+        // def f(a, (b,c), (d,e)):
+        // would expand to:
+        // def f(a, .1, .2):
+        //     (b, c) = .1
+        //     (d, e) = .2
+        AST_arguments* args;
+        if (source->ast->type == AST_TYPE::FunctionDef) {
+            args = ast_cast<AST_FunctionDef>(source->ast)->args;
+        } else {
+            args = ast_cast<AST_Lambda>(source->ast)->args;
+        }
+        int counter = 0;
+        for (AST_expr* arg_expr : args->args) {
+            if (arg_expr->type == AST_TYPE::Tuple) {
+                InternedString arg_name = source->getInternedStrings().get("." + std::to_string(counter));
+                AST_Name* arg_name_expr
+                    = new AST_Name(arg_name, AST_TYPE::Load, arg_expr->lineno, arg_expr->col_offset);
+                visitor.pushAssign(arg_expr, arg_name_expr);
+            }
```

Comment:
are Tuples the only things that can get this treatment?  a quick test seems to indicate it, but an assert that that the type is a tuple or a name could make that clear.